widget/tree.go

```diff
@@ -643,6 +643,12 @@ func (n *treeNode) Tapped(*fyne.PointEvent) {
 	n.tree.Select(n.uid)
 }
 
+func (n *treeNode) Resize(s fyne.Size) {
+	n.BaseWidget.Resize(s)
+	// TODO find out if the indentation changed. If not, and size is the same, then we can skip this as normal
```

Comment:
TBH, I think that a caller of `Resize` should never rely on the refresh. If the indentation gets changed at some point and depends on the refresh of a `Resize` call with the same size, this is wrong. Instead the change of the indentation should trigger a refresh.

Reply:
It's not relying on a Refresh - it's the Layout that we care about.
Normally we can ignore Resize if the size has not changed - but in the case of Tree it may be the same size but need to have the content moved when it's reused (because increased indent does not change size).
Maybe there is a better solution, but I cannot think of one right now.

Reply:
I still don’t fully understand why a `Resize` of a tree node has to handle indentation change. AFAICS an indentation change is implemented by the node’s `update` method which should then do the appropriate things.
Also, AFAIK, the `BaseWidget`'s `Resize` doesn’t do anything if the size didn’t change (apart from heating the world).

Reply:
Quite right, I managed to track down the root cause. The `leaf` and `branch` widgets were having the `treeNode` widget inside accessed directly, which was part of the issue. The rest was that the `partialRefresh` called when depth changed was not updating the node layout.

With this in place it should be fixed much more satisfactorily.